core/util.c

```diff
@@ -217,3 +217,48 @@ void register_print_function(print_message_function_t* function, int log_level)
     print_message_function = function;
     print_message_level = log_level;
 }
+
+/**
+ * Increment the mixed radix number by one according to the permutation matrix.
+ */
+void mixed_radix_incr(mixed_radix_int_t* mixed) {
+	int i = 0;
```

Comment:
Perhaps a check for `mixed` is warranted here and in the other function to make sure it is not null.